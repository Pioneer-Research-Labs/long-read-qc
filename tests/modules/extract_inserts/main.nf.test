nextflow_process {

    name "Test Process extract_inserts"
    script "modules/extract_inserts/main.nf"
    process "extract_inserts"

    test("Should extract inserts") {
        when {
            params {
                outdir = "tests/results"
            }
            process {
                """
                input[0] = Channel.fromList([ tuple( [id:'Gateway_EmptyBarcodedLib1_20241217',
                genome:'H_elongata'],
                file('${projectDir}/test_data/Gateway_EmptyBarcodedLib1_20241217/5HKJT6_1_sample_1.fastq.gz'),
                file('${projectDir}test_data/Gateway_EmptyBarcodedLib1_20241217/c.00323.dna'),
                file('${projectDir}/test_data/Gateway_EmptyBarcodedLib1_20241217/flanking.gb'))])
                """
            }
        }

        then {
            assert process.success
            assert snapshot("$launchDir/results/Gateway_EmptyBarcodedLib1_20241217/inserts.fasta").match()
        }
   }

}
