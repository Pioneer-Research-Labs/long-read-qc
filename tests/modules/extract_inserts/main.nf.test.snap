{
    "Should extract 0 inserts": {
        "content": [
            "inserts.fasta:md5,d41d8cd98f00b204e9800998ecf8427e"
        ],
        "meta": {
            "nf-test": "0.9.2",
            "nextflow": "25.04.1"
        },
        "timestamp": "2025-05-14T10:21:54.744428"
    },
    "Should extract some inserts": {
        "content": [
            "inserts.fasta:md5,7c0b0140289d271a6a5af323a69958e1",
            "cutadapt_inserts_report.json:md5,34a9e15e78e3d1156f91b0aaec57ba85",
            "untrimmed.fastq:md5,1be8d6feee12e1128cc857cee6d5537e",
            "cutadapt_info.tsv:md5,e40480ca30b0be410136bc499b0ea732"
        ],
        "meta": {
            "nf-test": "0.9.2",
            "nextflow": "25.04.2"
        },
        "timestamp": "2025-05-20T11:33:14.971386"
    },
    "Should extract inserts from a sample with no barcodes": {
        "content": [
            "inserts.fasta:md5,9ee3de61c7f497a93c60ca476f74381c"
        ],
        "meta": {
            "nf-test": "0.9.2",
            "nextflow": "25.04.1"
        },
        "timestamp": "2025-05-14T10:22:04.232316"
    }
}